components/aws/actions/dynamodb-execute-statement/dynamodb-execute-statement.mjs

```diff
@@ -1,57 +1,50 @@
-// legacy_hash_id: a_l0i7r1
-import AWS from "aws-sdk";
+import aws from "../../aws.app.mjs";
+import { toSingleLineString } from "../../common/utils.mjs";
 
 export default {
   key: "aws-dynamodb-execute-statement",
-  name: "DynamoDB - Execute Statement",
-  description: "This operation allows you to perform transactional reads or writes on data stored in DynamoDB, using PartiQL.",
-  version: "0.1.1",
+  name: "AWS - DynamoDB - Execute Statement",
+  description: toSingleLineString(`
+    This operation allows you to perform transactional reads or writes on data stored in DynamoDB, using PartiQL.
+    [See docs](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-dynamodb/classes/executestatementcommand.html)
+  `),
+  version: "0.1.2",
   type: "action",
   props: {
-    aws: {
-      type: "app",
-      app: "aws",
-    },
-    transact_statements: {
-      type: "any",
-      description: "Array < map > of PartiQL statements representing the transaction to run. Each element in the  array is composed of:\n* Statement - A PartiQL statment that uses parameters (required).\n* Parameters - Array <map>with the parameter values.",
-    },
+    aws,
     region: {
-      type: "string",
-      description: "The region to send service requests to.",
-      optional: true,
+      propDefinition: [
+        aws,
+        "region",
+      ],
     },
-    client_request_token: {
+    statement: {
       type: "string",
-      description: "Set this value to get remaining results, if `NextToken` was returned in the statement response.",
+      label: "Statement",
+      description: "The PartiQL statement representing the operation to run",
+    },
+    parameters: {
+      type: "string[]",
+      label: "Parameters",
+      description: "The parameters for the PartiQL statement, if any",
       optional: true,
     },
   },
-  async run() {
-    //See the API docs: https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/DynamoDB.html#executeTransaction-property
-
-    const {
-      accessKeyId,
-      secretAccessKey,
-    } = this.aws.$auth;
-
-    if (!this.transact_statements || !this.region) {
-      throw new Error("Must provide transact_statements, and region parameters.");
-    }
+  async run({ $ }) {
+    const params = {
+      Statement: this.statement,
+    };
 
-    const dynamodb = new AWS.DynamoDB({
-      accessKeyId,
-      secretAccessKey,
-      region: this.region,
-    });
+    if (this.parameters) params.Parameters = this.parameters;
```

Comment:
If `params.Parameters` is `undefined`, the request will **not** fail. I changed it to the following:
```
if (this.parameters?.length > 0) {
  const p = this.parameters.filter((param) => param);
  if (p.length > 0) params.Parameters = p;
}
```
This is because props that are an `array` of `strings` may contain empty strings; Pipdream doesn't strip them. If there are empty strings the API request will fail